elastic/client.go

```diff
@@ -501,3 +501,55 @@ func (p *ClientProvider) UpdateDocumentByQuery(index, query, fields string) ([]b
 
 	return resBytes, nil
 }
+
+// ReadWithScroll scrolls through the pages of size given in the query and adds up the scrollID in the result
+// Which is expected in the subsequent function call to get the next page, empty result indicates the end of the page
+func (p *ClientProvider) ReadWithScroll(index string, query map[string]interface{}, result interface{}, scrollID string) (err error)  {
+	var res *esapi.Response
+	defer func() {
+		if err := res.Body.Close(); err != nil {
+			log.Printf("Err: %s", err.Error())
+		}
+	}()
+
+	if scrollID == "" {
+		var buf bytes.Buffer
+		err = json.NewEncoder(&buf).Encode(query)
+		if err != nil {
+			return err
+		}
+
+		res, err = p.client.Search(
+			p.client.Search.WithIndex(index),
+			p.client.Search.WithBody(&buf),
+			p.client.Search.WithScroll(time.Minute),
+		)
+	} else {
+		res, err = p.client.Scroll(p.client.Scroll.WithScrollID(scrollID), p.client.Scroll.WithScroll(time.Minute))
+	}
+	if err != nil {
+		return err
+	}
+	if res.StatusCode == 200 {
```

Comment:
I think we can use constant for all http status codes.

Reply:
done.